### PR TITLE
Multiple repository url

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**
Helm broker can be configured with many URLs to bundle repositories

Changes proposed in this pull request:
- configuration with many repo URLs
- Helm-Broker triggers Service Catalog to sync when Helm-Broker starts (or restarts). Because of DNS propagation takes some time, before triggering a sync, Helm Broker checks if the DNS is refreshed - do a call to http://core-helm-broker/statusz and expects HTTP 200
- Helm-Broker reads bundles from repomte repositories when the Service Catalog calls to 'catalog' endpoint

Running the Helm-Broker in Kyma needs changes in the chart:
 - add and define envs
 -- APP_REPOSITORY_URLS 
     value: {{ .Values.repository.URLs }}
 -- APP_CLUSTER_SERVICE_BROKER_NAME
     core-helm-broker
 -- HelmBrokerURL APP_HELM_BROKER_URL
     https://core-helm-broker
- values.yaml
repository:
  URLs: "https://github.com/kyma-project/bundles/releases/download/0.1.0/"

set proper RBAC

- apiGroups: ["servicecatalog.k8s.io"]
  resources: ["clusterservicebrokers"]
  verbs: ["get","update"]

deployment change (consider RollingUpdate):
strategy:
    type: Recreate

**Related issue(s)**

### CODE REVIEW

File: components/helm-broker/internal/bundle/syncer.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+//go:generate mockery -name=Provider -output=automock -outpkg=automock -case=underscore
+
+type bundleOperations interface {
+	FindAll() ([]*internal.Bundle, error)
+	RemoveByID(internal.BundleID) error
+	Upsert(*internal.Bundle) (replace bool, err error)
+}
+
+// Provider contains method which provides CompleteBundle items.
+type Provider interface {
```
Comment (polskikiel) [inline components/helm-broker/internal/bundle/syncer.go]: Why do you export that interface?

File: components/helm-broker/internal/broker/catalog_syncer.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	osb "github.com/pmorie/go-open-service-broker-client/v2"
+)
+
+//go:generate mockery -name=syncer -output=automock -outpkg=automock -case=underscore
+type syncer interface {
+	Execute()
+}
+
+type catalogSyncerService struct {
+	underlying catalogGetter
```
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply components/helm-broker/internal/broker/catalog_syncer.go]: `underlying` is a common pattern for a name (sometimes developers use also `target`) in such wrapper or layer pattern implementation.

File: components/helm-broker/cmd/broker/main.go [hunk 3]
```diff
@@ -4,75 +4,120 @@ import (
...
+		case <-timeoutCh:
+			log.Errorf("Waiting for service %s to be ready timeout %s exceeded.", url, timeout.String())
+			if err != nil {
+				log.Errorf("Last call error: %s", err.Error())
+			} else {
+				log.Errorf("Last call response status: %s", r.StatusCode)
+			}
+			break
+		default:
+			time.Sleep(2 * time.Second)
```
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply components/helm-broker/cmd/broker/main.go]: It takes about 30 seconds, let's decrease it to one second

File: components/helm-broker/internal/bundle/syncer.go (hunk 1 above)
Comment (piotrmiskiewicz) [reply components/helm-broker/internal/bundle/syncer.go]: unexported

File: components/helm-broker/README.md [hunk 4]
```diff
@@ -18,3 +18,13 @@ You need the following tools to set up the project:
...
+
+### Run a local version
+
+To run the application without building the binary file, run this command:
+
+```bash
+APP_KUBECONFIG_PATH=/Users/$User/.kube/config APP_CONFIG_FILE_NAME=contrib/minimal-config.yaml APP_REPOSITORY_URLS=https://github.com/kyma-project/bundles/releases/download/0.1.0/ APP_CLUSTER_SERVICE_BROKER_NAME=core-helm-broker APP_HELM_BROKER_URL=http://localhost:8080 go run cmd/broker/main.go
+```
+
+Be aware when you run the Helm Broker locally you cannot test all features. Only part of them are working correctly.
```
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply components/helm-broker/README.md]: added a comment

File: components/helm-broker/internal/broker/broker_syncer_test.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+	// then
+	sb, err := client.ServicecatalogV1beta1().ClusterServiceBrokers().Get(fixClusterServiceBroker().Name, v1.GetOptions{})
+	require.NoError(t, err)
+
+	assert.Equal(t, int64(1), sb.Spec.RelistRequests)
+	assert.Nil(t, err)
+}
+
+func TestServiceBRokerSync_NotExistingBroker(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply components/helm-broker/internal/broker/broker_syncer_test.go]: fixed

File: components/helm-broker/internal/broker/broker_syncer.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+		// do not return error if the broker does not exists, the method is dedicated to update
+		// ClusterServiceBroker resource if exists. If it is not created yet
+		// - it will be created in the future and Service Catalog will call the 'catalog' endpoint soon.
+		case apiErrors.IsNotFound(err):
+			return nil
+		case err != nil:
+			return errors.Wrapf(err, "while getting ClusterServiceBrokers %s", name)
+		}
+
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply components/helm-broker/internal/broker/broker_syncer.go]: removed

File: components/helm-broker/README.md [hunk 7]
```diff
@@ -18,3 +18,13 @@ You need the following tools to set up the project:
...
+
+### Run a local version
+
+To run the application without building the binary file, run this command:
+
+```bash
+APP_KUBECONFIG_PATH=/Users/$User/.kube/config APP_CONFIG_FILE_NAME=contrib/minimal-config.yaml APP_REPOSITORY_URLS=https://github.com/kyma-project/bundles/releases/download/0.1.0/ APP_CLUSTER_SERVICE_BROKER_NAME=core-helm-broker APP_HELM_BROKER_URL=http://localhost:8080 go run cmd/broker/main.go
+```
+
+Be aware when you run the Helm Broker locally you cannot test all features. Only part of them are working correctly. All features which performs an action with Tiller won't work.
```
Comment (klaudiagrz) [inline components/helm-broker/README.md]:
```suggestion
>**NOTE:**  Not all features are available when you run the Helm Broker locally. All features which perform actions with Tiller do not work.
```

File: docs/service-brokers/docs/011-configuration-helm-broker.md [hunk 8]
```diff
@@ -33,24 +33,25 @@ Follow these steps to change the configuration and make the Helm Broker fetch bu
 
     - A `{bundle_name}-{bundle_version}.tgz` file for each bundle version defined in the `yaml` file. The `.tgz` file is an archive of your bundle's directory.
 
-2. In the `values.yaml` provide your server's URL in the **repository.URL** attribute:
+2. In the `values.yaml` provide your server's URLs in the **repository.URLs** attribute as a list of URLs separated by a semi-colon:
```
Comment (klaudiagrz) [inline docs/service-brokers/docs/011-configuration-helm-broker.md]:
```suggestion
2. In the `values.yaml` provide your server's URLs in the **repository.URLs** attribute as a list of URLs separated by a semicolon:
```

File: docs/service-brokers/docs/011-configuration-helm-broker.md [hunk 9]
```diff
@@ -33,24 +33,25 @@ Follow these steps to change the configuration and make the Helm Broker fetch bu
...
 3. Install Kyma on Minikube. See the **Install Kyma locally from the release** document to learn how.
 
-### Configure repository URL in the runtime
+### Configure repository URLs in the runtime
 
-Follow these steps to change the repository URL:
+Follow these steps to add the repository URL:
 
-1. Configure Helm Broker:
+1. Add new bundle repository URL:
```
Comment (klaudiagrz) [inline docs/service-brokers/docs/011-configuration-helm-broker.md]:
```suggestion
1. Add a new bundle repository URL:
```

File: docs/service-brokers/docs/011-configuration-helm-broker.md [hunk 10]
```diff
@@ -59,8 +60,5 @@ Follow these steps to change the repository URL:
  kubectl get pod -n kyma-system -l app=core-helm-broker
  ```
 
-3. Trigger the Service Catalog synchronization:
+Running Helm Broker triggers the Service Catalog synchronization automatically. New Service Classes should be after a half of minute.
```
Comment (klaudiagrz) [inline docs/service-brokers/docs/011-configuration-helm-broker.md]:
```suggestion
Running the Helm Broker triggers the Service Catalog synchronization automatically. New ServiceClasses appear after a half-minute.
```

**Pull request merged**